framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -89,7 +91,12 @@
         if (AnnotationUtils.areSameByClass(anno, DoubleVal.class)) {
             values = convertDoubleVal(anno, castType, castTo);
         } else if (AnnotationUtils.areSameByClass(anno, IntVal.class)) {
-            values = convertIntVal(anno, castType, castTo);
+            List<Long> longs = ValueAnnotatedTypeFactory.getIntValues(anno);
+            values = convertIntVal(longs, castType, castTo);
+        } else if (AnnotationUtils.areSameByClass(anno, IntRange.class)) {
+            Range range = ValueAnnotatedTypeFactory.getIntRange(anno);
+            List<Long> longs = getValuesFromRange(range, Long.class);
```

Comment:
There is no javadoc for this existing method, so it is not clear to me when `getValuesCastedToType` is used.
All other uses of `getValuesFromRange` seem to be guarded by MAX_VALUES. What happens here if the Range is very large? Would it create a List with all those value? We should make sure that we never enumerate a Range class if that would encompass more than MAX_VALUES, to prevent large memory consumption and time in the compiler.